packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_view.dart

```diff
@@ -4,32 +4,61 @@
 
 import 'package:flutter/material.dart';
 
+import '../../../../../shared/common_widgets.dart';
+import '../../../../../shared/split.dart';
+import '../../../shared/heap/model.dart';
+import '../controller/diff_pane_controller.dart';
 import '../controller/model.dart';
+import 'class_details.dart';
 import 'stats_table.dart';
 
 class SnapshotView extends StatelessWidget {
-  const SnapshotView({Key? key, required this.item}) : super(key: key);
+  const SnapshotView({Key? key, required this.controller}) : super(key: key);
 
-  final SnapshotListItem item;
+  final DiffPaneController controller;
 
   @override
   Widget build(BuildContext context) {
+    final item = controller.selectedItem as SnapshotListItem;
     return ValueListenableBuilder<bool>(
       valueListenable: item.isProcessing,
       builder: (_, isProcessing, __) {
         if (isProcessing) return const SizedBox.shrink();
 
-        final stats = item.heap?.stats;
+        late HeapStatistics? stats;
+        if (item.diffWith.value == null) {
+          stats = item.heap?.stats;
+        } else {
+          final heap1 = item.heap!;
+          final heap2 = item.diffWith.value!.heap!;
+          stats = controller.diffStore.compare(heap1, heap2).stats;
```

Comment:
if this can be expensive, we shouldn't do this in build. If it is expensive, we could do this in the controller before we set isProcessing to false (that way we still show a loading spinner while we are calculating this).

Reply:
It is ok for small applications. 
And it may be ok for performance, but heavy for memory.
I need to experiment with Gallery. 
Added to the issue so that we do not forget: https://github.com/flutter/devtools/issues/3949#issue-1190345467

Reply:
If this has the potential to slow down the UI at all, we should definitely ship it out to a future and have the UI with a future builder and a spinner or something :)

Reply:
ok

Added TODO and item to https://github.com/flutter/devtools/issues/3949#issue-1190345467